content/en/blog/2020/workload-entry/index.md

```diff
@@ -0,0 +1,100 @@
+
+Istio 1.6 has introduced a few changes in how you manage non-Kubernetes workloads, driven by a desire to make it easier to gain Istio's benefits for use cases beyond containers, such as running traditional databases on a platform outside of Kubernetes, or adopting Istio's features for existing applications without rewriting them.
+
+### Background
+
+Prior to Istio 1.6, non-containerized workloads were configurable simply as an IP address in a `ServiceEntry`, which meant that they only existed as part of a service. Istio lacked a first-class abstraction for these non-containerized workloads, something similar to how Kubernetes treats Pods as the fundamental unit of compute - a named object that serves as the collection point for all things related to a workload - name, labels, security properties, lifecycle status events, etc. Enter `WorkloadEntry`.
+
+Consider the following `ServiceEntry` describing a service implemented by a few tens of VMs with IP addresses:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: svc1
+spec:
+  hosts:
+  - svc1.internal.com
+  ports:
+  - number: 80
+    name: http
+    protocol: HTTP
+  resolution: STATIC
+  endpoints:
+  - address: 1.1.1.1
+  - address: 2.2.2.2
+  ....
+{{< /text >}}
+
+If you wanted to migrate this service into Kubernetes in an active-active manner - i.e. launch a bunch of Pods, send a portion of the traffic to the Pods over Istio mutual TLS (mTLS) and send the rest to the VMs without sidecars - how would you do it? You would have needed to use a combination of a Kubernetes service, a virtual service, and a destination rule to achieve the behavior. Now, let's say you decided to add sidecars to these VMs, one by one, such that you want only the traffic to the VMs with sidecars to use Istio mTLS. If any other Service Entry happens to include the same VM in its addresses, things start to get very complicated and error prone.
+
+The primary source of these complications is that Istio lacked a first-class definition of a non-containerized workload, whose properties can be described independently of the service(s) it is part of.
+
+{{< image
+    link="./workload-entry-first-example.svg"
+    alt="Service Entries Pointing to Workload Entries"
+    caption="The Internal of Service Entries Pointing to Workload Entries"
+    >}}
+
+### Workload Entry: A Non-Kubernetes Endpoint
+
+`WorkloadEntry` was created specifically to solve this problem. `WorkloadEntry` allows you to describe non-Pod endpoints that should still be part of the mesh, and treat them the same as a Pod. From here everything becomes easier, like enabling `MUTUAL_TLS` between workloads, whether they are containerized or not.
+
+To create a `[WorkloadEntry](/docs/reference/config/networking/workload-entry/)` and attach it to a `[ServiceEntry](/docs/reference/config/networking/service-entry/)` you can do something like this:
```

Comment:
FYI these links are not working properly on the live site, I think the backticks are messing it up?

Reply:
Yew, the backticks should be inside the square brackets around the word WorkloadEntry only.

Reply:
Ah good catch, I've pushed: [this PR](https://github.com/istio/istio.io/pull/7381) to fix this. It doesn't make them look like links which is far from ideal, but it at least still links the text. This also seems to be what other articles do, so we're in line even if it's not the greatest.